src/routes/Settings/SettingsIndexViewContainer.js

```diff
@@ -1,5 +1,28 @@
 // @flow
+import { get } from 'lodash'
+import { connect } from 'react-redux'
 
 import SettingsIndexView from './SettingsIndexView'
+import { selectActiveWallet } from '../../store/selectors/wallets'
+import { selectCurrentNetwork } from '../../store/selectors/networks'
 
-export default SettingsIndexView
+function mapStateToProps(state) {
+  const networkName = get(selectCurrentNetwork(state), 'title', null)
+
+  const activeWallet = selectActiveWallet(state)
+  const walletName = get(activeWallet, 'name', null)
+  const walletType = get(activeWallet, 'customType', null)
+  const { derivationPath, passphrase } = get(activeWallet, 'mnemonicOptions', {})
+
+  const resultProps = {
+    networkName,
+    walletName,
+    walletType,
+    derivationPath,
+    passphrase,
+  }
+
+  return { ...state.settings, ...resultProps } || {}
+}
```

Comment:
state.settings better to get with stateSelector and pass to result objects as `settings`
so in View you will use in props:
settings: SettingsState instead of ...SettingsState

Reply:
you can pass all `resultProps` in return value without creating of useless variable

Reply:
why we need ` || {}` such fallback